client/service_create.go

```diff
@@ -98,6 +98,11 @@ func imageDigestAndPlatforms(ctx context.Context, cli DistributionAPIClient, ima
 	if len(distributionInspect.Platforms) > 0 {
 		platforms = make([]swarm.Platform, 0, len(distributionInspect.Platforms))
 		for _, p := range distributionInspect.Platforms {
+			// clear architecture field for arm. This is a temporary fix.
+			arch := p.Architecture
+			if arch == "arm" || arch == "Arm" {
```

Comment:
if `stringsToLower(arch) == "arm"`

Reply:
Also should this be `HasPrefix`?

Reply:
In the images I've seen, it's only been "arm". We could use `HasPrefix` to be extra careful but I don't think it's necessary.